File: Janus/app/schemas/instance.py
"""Schemas for Bergamot instance discovery and feature negotiation."""

from pydantic import BaseModel


class ProductDescriptor(BaseModel):
    name: str
    tagline: str


class ServiceEndpoints(BaseModel):
    http_base_url: str
    api_base_url: str
    websocket_base_url: str
    uploads_base_url: str
    media_base_url: str | None = None
    admin_base_url: str | None = None
    livekit_base_url: str | None = None


class RuntimeConfigRead(BaseModel):
    manifest_version: int
    product: ProductDescriptor
    services: ServiceEndpoints


class FeatureManifestRead(BaseModel):
    manifest_version: int
    product: ProductDescriptor
    capabilities: dict[str, bool]
    routes: dict[str, str]

File: packages/contracts/src/instance.ts
import manifest from "../data/instance-manifest.json";
import type { BergamotCapabilityFlags } from "./capabilities";

export interface BergamotProductDescriptor {
  name: string;
  tagline: string;
}

export interface BergamotServiceEndpoints {
  httpBaseUrl: string;
  apiBaseUrl: string;
  websocketBaseUrl: string;
  uploadsBaseUrl: string;
  mediaBaseUrl: string | null;
  adminBaseUrl: string | null;
  livekitBaseUrl: string | null;
}

export interface BergamotRuntimeConfig {
  manifestVersion: number;
  product: BergamotProductDescriptor;
  services: BergamotServiceEndpoints;
}

export interface BergamotRouteManifest {
  login: string;
  register: string;
  dmHome: string;
  notifications: string;
  favorites: string;
}

export interface BergamotFeatureManifest {
  manifestVersion: number;
  product: BergamotProductDescriptor;
  capabilities: BergamotCapabilityFlags;
  routes: BergamotRouteManifest;
}

export const defaultProductDescriptor = manifest.product as BergamotProductDescriptor;
